tools/debugging/stress_test_transfers.py

```diff
@@ -0,0 +1,729 @@
+                    if result.get() != STATUS_CODE_FOR_SUCCESS:
+                        self.stop.set()
+
+                process.result.rawlink(subprocess_stopped)
+
+        return ProcessNursery()
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_value: Optional[BaseException],
+        traceback: Optional[TracebackType],
+    ) -> Optional[bool]:
+        # Make sure to signal that we are exiting. This is a noop if the signal
+        # is set already (e.g. because a subprocess exited with a non-zero
+        # status code)
+        self.stop.set()
+
+        # Behave nicely if context manager's __exit__ is executed. This
+        # implements the expected behavior of a context manager, which will
+        # clear the resources when exiting.
+        atexit.unregister(self._free_resources)
+
+        self._free_resources()
+
+        return None
+
+    def _free_resources(self) -> None:
+        for p in self._processes:
+            p.send_signal(signal.SIGINT)
+
+
+@lru_cache()
+def factorize_number(number: int) -> Iterator[int]:
+    """Factorize `number` into its primes.
+
+    Note:
+        This function skips the number 1.
+        This function yields each prime `n` times, where `n` is how many times
+        it can be used to devide `number`.
+    """
+    if number <= 1:
+        raise ValueError("Cannot factorize 1, 0, or negative numbers.")
+
+    # Special case 2 to avoid emitting the same value twice
+    if number == 2:
+        yield 2
+        return
+
+    # Optimization: Do the even numbers on a single step, this reduces the
+    # number of candidates in half.
+    while number % 2 == 0:
+        yield 2
+        number //= 2
+
+    # `number` was even
+    if number == 1:
+        return
+
+    # There cannot be a prime that is larger than `sqrt(number)`, this upper
+    # bound drastically reduces the search space.
+    upper_bound = sqrt(number)
+
+    # The `+1` covers the corner case were the `sqrt` itself is a factor. E.g.
+    # `sqrt(9)` is a factor of `9`, and `range(3,3)` would skip it.
+    upper_bound = ceil(upper_bound) + 1
+
+    # skip all even numbers, since they have been processed above
+    step = 2
+    candidates = range(3, upper_bound, step)
+
+    for candidate in candidates:
+        while number % candidate == 0:
+            yield candidate
+            number //= candidate
+
+    # The last number is also a prime
+    if number != 1:
+        yield number
+
+
+def all_proper_divisors(number: int) -> Iterator[int]:
+    """Returns all integers that are proper divisors of `number`.
+
+    A proper `divisor` is a number will divides `number` evenly. i.e. `number %
+    divisor == 0`.
+
+    Note:
+        This does not return the number `1`, otherwise some of the proper
+        divisors would be repeated.
+    """
+    if number <= 1:
+        raise ValueError("number most be greater than 0")
```

Comment:
message does not match the condition (small typo got through as well)

Reply:
ahhaha, thanks @BOR4 :)